rfcs/0001-tokens-naming-convention.md

```diff
@@ -0,0 +1,495 @@
+
+> At what level(s) do I store this purposeful decision? Is it ok to store the same decision in two different locations? If purpose of two different choices is nearly identical, should it be 1 or 2 tokens?
+
+Both `color.feedback` and `Notification.color.text` concepts have other variants (`warning`, `success`, `info` and `label`, `value`, and `helper-text`, respectively) for which `error` completes a set. Even if the actual red value is the same, we would consider aliasing the specific one to the generic one
+
+```js
+// specific = generic
+Notification.color.text.error = color.feedback.error
+```
+This also gives us flexibility to change the `Notification.color.text.error` color anytime without impacting other uses of `color.feedback.error`.
+
+## Principles Used
+There are certain principles that were kept in mind while creating this naming framework
+### Specificity over Flexibility
+Tokens like `theme.color.success` combine *category* (`color`) and *variant* (`success`) but leaves the interpretation to the user to apply `theme.color.success` to any of `background`, `border` or `text`.
+
+Flexibility comes at the expense of specificity. A `success` color may only be intended for `text` or `background` but not both. Even more, an object reflecting `success` may require distinct colors for `text` versus `background` versus `border`. In this case, including a *property* level in a token results in a more specific yet less flexiblity eg: `theme.color.background.success` or `theme.color.text.success`.
+
+### Start within, then promote across
+
+Start with making the tokens `local` to the components, then follow the usage pattern, if used more than twice then promote it to `global` token. This is a healthy way to add tokens gradually without adding things blindly to local components or globally.
+### Theme ≠ Mode
+
+A theme may eventually require `light`, `dark` modes. PG might require a `light` and `dark` mode but their theme is different from Razorpay X. Similarly, Razorpay X may require `dark` and `light` mode but their dark and light could be different from any other product's theme
+## How/Where will we store these tokens?
+### Storing theme tokens
+The theme tokens shall we created under `packages/blade/themes`
+### Storing component tokens
+The component tokens can be created in 2 ways:
+1. Co-located inside the component file.
+2. Create a token file under the component's directory i.e `packages/blade/components/Button/tokens.ts`.
+3. Create component tokens under `packages/blade/tokens`.
+
+Until now I believe that if we store the component level tokens under the components directory that'll be better because:
+1. We can refer them from a central token file in all the platform level implementation of a particular component(React/React Native/Svelte etc.)
```

Comment:
"tokens under the components" is not "central".
Central is the 3rd approach.

Reply:
Another difference with 2nd and 3rd approach is tokens would be stored in separate packages. And version for these packages might be different as we had discussed earlier.
Would we be enforcing people to upgrade to latest theme every time they want to upgrade even a single component package? How would we maintain backward compatibility?

Reply:
Oh, this is also there in open questions. Saw it now. 🙈